test/e2e/framework/podwaits/podwaits.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podwaits
+
+import (
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+// WaitForPodRunning waits for the pod to run in the namespace.
+func WaitForPodRunning(f *framework.Framework, podName string) error {
+	return e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, podName, f.Namespace.Name)
```

Comment:
We should move this into the e2epod package and refactor it to use a clienset and a namespace instead of extracting it itself from the framework - to allow people to use it as they wish and to avoid the circular dependency in e2epod.
An example of a function like this one https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pod/wait.go#L105

```suggestion
func WaitForPodRunning(c clientset.Interface, ns string podName string) error {
	return WaitForPodNameRunningInNamespace(c, podName, ns)
```
The above (i think it should work) can now be included with e2epod.
All calling functions just need to be updated to use it
```go
e2epod.WaitForPodRunning(f.ClientSet, f.Namespace.Name, PodName)
```

Reply:
Yeah, that is a nice point.
I did the same thing by implementing it in e2epod, but that made circular dependency.
Then I think it is nice to move this function into separated package e2epodwaits.
Ideally e2epod is the best, but e2epodwaits is not so bad at this time I feel.

Reply:
What caused the circular dependency in e2epod ?
The only place within the framework where I see this being used in in  test/e2e/framework/framework.go.
If anything in the framework is using that function then we should move it to a subpkg as well.

If we can we should avoid making temp packages like e2epodwaits.
It can make code reviewing easier for us but there is the possibility that people are writing/maintaining their e2e tests and having code be shifted may be confusing.
If we can avoid a temporary subpkg we should, unless there it requires a lot of refactoring work to do so.

Reply:
> What caused the circular dependency in e2epod ?

Because `e2e/framework/util.go` uses e2epod as https://github.com/kubernetes/kubernetes/blob/b6f11382f5d63313b968028719e54a1afb3fe1ba/test/e2e/framework/util.go#L81
That is an invalid dependency(core framework import sub frameworks) as you know, and it is hard to solve it at this time.
In addition, the moved function depends on core framework code as this PR.
So if moving the function into e2epod, circular dependency happens like
- core framework `util.go` imports e2epod (wrongly)
- e2epod becomes importing core framework

> If we can we should avoid making temp packages like e2epodwaits. It can make code reviewing easier for us but there is the possibility that people are writing/maintaining their e2e tests and having code be shifted may be confusing. If we can avoid a temporary subpkg we should, unless there it requires a lot of refactoring work to do so.

Yes, I totally agree with you.
TBH, I am not sure this e2epodwaits is temporary workaround or not because there are already similar subpackages like e2epod and e2epodlogs.

Reply:
> In addition, the moved function depends on core framework code as this PR.

To solve that, `WaitForPodRunning` can be rewritten to not import the framework and instead take a clientset and a namespace as arguments
```go
func WaitForPodRunning(c clientset.Interface, ns string podName string) error {
```

> Because e2e/framework/util.go uses e2epod as

Looking at this PR, the e2e framework doesn't make use of `WaitForPodRunning` so it would not cause a circular dependency. If it was rewritten (I think).
`WaitForPodRunning` depends on code that is well contained within e2epod
* https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pod/wait.go#L320

---

With respect to fixing the fact that the framework imports e2epods, we need to refactor some more code from utils to not have the framework importing it and that is an AI.
Whenever that gets done it would also be an AI to merge e2epodwaits with e2epod.
I think that if we can do it now there is no reason to leave it for later.
Then we can just focus on removing the e2epod import from framework/util.go

also, just looked at e2epodlogs and I think that one would also be easy to consolidate within e2epod (as an aside thought).

Reply:
@alejandrox1 

> To solve that, WaitForPodRunning can be rewritten to not import the framework and instead take a clientset and a namespace as arguments

That seems a nice idea, thanks! :-)
OK, let me try it.

Reply:
We just need to replace these calls with the existing `e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, podName, f.Namespace.Name)` of e2epod package without moving the function because of specifying the namespace.

Reply:
@alejandrox1 
> also, just looked at e2epodlogs and I think that one would also be easy to consolidate within e2epod (as an aside thought).

I did the above with https://github.com/kubernetes/kubernetes/pull/89180